### ISSUE TITLE
Is it possible to suppose multiple versions of HTML5 games?

### ISSUE THREAD

**Issue by TerryCavanagh** [itchio/itch.io#869]
Here's a thing I'd love to do with my game [Dicey Dungeons](https://terrycavanagh.itch.io/dicey-dungeons): Make the old versions of the game playable by clicking on them in the list of versions, here:

![image](https://user-images.githubusercontent.com/2874150/43717399-5a260db0-997f-11e8-8c1d-722bfafd7d97.png)

Also, potentially, I could make different versions of the game available this way - for example, the current version of the game is using huge 4k art, but this won't work on certain old devices. I'd love to be able to provide two versions - a 4k one and a 1080p one, and let people pick!

Just wondering if you've got anything like this planned!

**Comment by leafo** [comment]
I'm trying to think of the easiest way to introduce this. The main complication is that the UI on the edit game page essentially built around the idea of selecting 1 embed, not many: You set the type of the page, then you can only pick one upload to be embedded.

I would need a new way to select what to embed, along with allowing other things to still be marked as HTML5 games. Additionally, the embed settings are globally set, so that UI might have to change.

And then, the game page's UI would be able to handle the case when there are multiple things uploaded.

**Comment by TerryCavanagh** [comment]
Ah! Well, no worries, this is definitely a "wouldn't it be nice if" feature rather than something that's actually important :) It does sound complicated!

**Comment by fasterthanlime** [comment]
@leafo Nope, in v25, if there's no native executables (just an HTML5 file), then it launches it as a web game.

Too many folks were just uploading an index.html / zip with an index.html file without tagging as HTML5, so I just decided to support that too.

**Comment by TerryCavanagh** [comment]
Just out of curiosity, if I upload a zip file with different versions of the project in different folders, what happens on the users end? Does the zip get decompressed on itch's servers, or does the user download the whole thing first? (I'm thinking of making a simple HTML page to choose between different versions, and putting the whole lot in one big zip).

(related: https://github.com/itchio/itch.io/issues/870)

**Comment by fasterthanlime** [comment]
Going through old threads,

> Does the zip get decompressed on itch's servers

Yes, it does, so the approach you mention above is sound.

**Comment by Goury** [comment]
Bump.
@leafo any progress on this one?
If you need, I can help.
